Execute operation order + operation order validation + java version migration to improve JavaFX support

dilumdesilva opened the pull request:
## Description
operations order execution mechanism and operations adding validation logic mechanism This commit includes implementation related to the operation execution according to the order of dragged components order. Also, the validation logic to check which elements can be dragged one before/after to a particular element.

**Known Issues**
This change contains known issues when executing the OpenCV operation order.

Also, this version includes a java JDK 11 --> 8 migration and Lombok support dependencies 

## Type of change 
- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Current Status 
<img width="1281" alt="Screenshot 2020-07-26 at 21 17 21" src="https://user-images.githubusercontent.com/23298579/88483420-73e1cf80-cf85-11ea-8004-7612271a3ab6.png">

nvinayvarma189 commented:
Hey, I'm new to this repo and I think instead of making these changes, we can put the `app.py` file outside the src folder? @samagragupta @AnjaliAstro

dilumdesilva committed:
Add: operations order execution mechanism + operations adding validation logic mechanism
This commit includes implementation related to the operation execution according to the order of dragged components order. Also, the validation logic to check which elements can be dragged one before/after to a particular element.

Also this commit has a migration to java jdk 8

Also, contains known issues when executing the openCV operation order.

Add: default image to use for the processing operations

dilumdesilva committed:
Fix: operation queue logic
